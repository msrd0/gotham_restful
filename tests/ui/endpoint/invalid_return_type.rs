#[macro_use]
extern crate gotham_restful;
use gotham_restful::gotham::hyper::Method;

#[derive(Resource)]
#[resource(endpoint)]
struct FooResource;

struct FooResponse;

#[endpoint(method = "Method::GET", uri = "")]
fn endpoint() -> FooResponse {
	unimplemented!()
}

fn main() {}
