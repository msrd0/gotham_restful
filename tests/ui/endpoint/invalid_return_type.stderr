error[E0277]: the trait bound `FooResponse: ResponseSchema` is not satisfied
  --> tests/ui/endpoint/invalid_return_type.rs:12:18
   |
12 | fn endpoint() -> FooResponse {
   |                  ^^^^^^^^^^^ the trait `ResponseSchema` is not implemented for `FooResponse`
   |
   = help: the following other types implement trait `ResponseSchema`:
             NoContent
             Pin<Box<(dyn Future<Output = Res> + Send + 'static)>>
             Raw<T>
             Redirect
             Result<NoContent, E>
             Result<R, E>
             Result<Raw<T>, E>
             Result<Redirect, E>
             Success<T>
note: required by a bound in `gotham_restful::EndpointWithSchema::Output`
  --> src/endpoint.rs
   |
   |     #[openapi_bound("Output: crate::ResponseSchema")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Output`

error[E0277]: the trait bound `FooResponse: gotham_restful::IntoResponse` is not satisfied
  --> tests/ui/endpoint/invalid_return_type.rs:12:18
   |
12 | fn endpoint() -> FooResponse {
   |                  ^^^^^^^^^^^ the trait `gotham_restful::IntoResponse` is not implemented for `FooResponse`
   |
   = help: the following other types implement trait `gotham_restful::IntoResponse`:
             NoContent
             Pin<Box<(dyn Future<Output = Res> + Send + 'static)>>
             Raw<T>
             Redirect
             Result<NoContent, E>
             Result<R, E>
             Result<Raw<T>, E>
             Result<Redirect, E>
           and 2 others
note: required by a bound in `gotham_restful::EndpointWithSchema::Output`
  --> src/endpoint.rs
   |
   |     type Output: IntoResponse + Send;
   |                  ^^^^^^^^^^^^ required by this bound in `gotham_restful::EndpointWithSchema::Output`
